agent/src/main/java/org/openjdk/jmc/agent/sap/boot/util/Arguments.java

```diff
@@ -0,0 +1,269 @@
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+package org.openjdk.jmc.agent.sap.boot.util;
+
+import java.util.HashMap;
+import java.util.regex.Pattern;
+
+public class Arguments {
+
+	private final HashMap<String, String[]> args;
+	private final Command command;
+	private volatile Object customData;
+
+	Arguments(String optionsLine) {
+		this.command = null;
+		this.args = getOptions(optionsLine);
+	}
+
+	Arguments(String line, Command command) {
+		this.command = command;
+		this.args = getOptions(line);
+	}
+
+	public Command getCommand() {
+		return command;
+	}
+
+	public Object getCustomData() {
+		return customData;
+	}
+
+	public void setCustomData(Object customData) {
+		this.customData = customData;
+	}
+
+	public boolean hasOption(String option) {
+		return args.containsKey(option);
+	}
+
+	public boolean hasHelpOption() {
+		return args.containsKey("help");
+	}
+
+	public String getString(String option, String defaultResult) {
+		if (args.containsKey(option)) {
+			String[] opts = args.get(option);
+
+			if (opts.length == 0) {
+				return null;
+			}
+
+			if (opts.length == 1) {
+				return opts[0];
+			}
+
+			reportOptionError(option, "Found " + opts.length + " values for option. Expected only one.");
+		}
+
+		return defaultResult;
+	}
+
+	public String[] getStrings(String option) {
+		if (args.containsKey(option)) {
+			return args.get(option);
+		}
+
+		return new String[0];
+	}
+
+	public boolean getBoolean(String option, boolean defaultResult) {
+		if (args.containsKey(option)) {
+			return Boolean.parseBoolean(getString(option, ""));
+		}
+
+		return defaultResult;
+	}
+
+	public Pattern getPattern(String option, Pattern defaultResult) {
+		if (args.containsKey(option)) {
+			return Pattern.compile(getString(option, ""));
+		}
+
+		return defaultResult;
+	}
+
+	public int getInt(String option, int defaultRersult) {
+		if (args.containsKey(option)) {
+			try {
+				return Integer.parseInt(getString(option, ""));
+			} catch (NumberFormatException e) {
+				reportOptionError(option, "Could not parse integer value");
+			}
+		}
+
+		return defaultRersult;
+	}
+
+	public String getUnknownArgument() {
+		for (String key : args.keySet()) {
+			if (!command.hasOption(key)) {
+				return key;
+			}
+		}
+
+		return null;
+	}
+
+	public long getLong(String option, long defaultRersult) {
+		if (args.containsKey(option)) {
+			try {
+				return Long.parseLong(getString(option, ""));
+			} catch (NumberFormatException e) {
+				reportOptionError(option, "Could not parse integer value");
+			}
+		}
+
+		return defaultRersult;
+	}
+
+	public double getDouble(String option, double defaultRersult) {
+		if (args.containsKey(option)) {
+			try {
+				return Double.parseDouble(getString(option, ""));
+			} catch (NumberFormatException e) {
+				reportOptionError(option, "Could not parse integer value");
```

Comment:
Could not parse Double value? Or is it really int here too ?

Reply:
No, the message was wrong.